### PR TITLE
Add type hints for mmcv/ops

### PR DESCRIPTION
**Pull request by ShawnHXH**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add type hints for files in mmcv/ops.

## Modification

Add type hints for files in mmcv/ops:
- ball_query.py
- border_align.py
- correlation.py
- deform_conv.py,
- deform_roi_pool.py
- deprecated_wrappers.py [**NO type hints need to be added in this file**]

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Review by HAOCHENYE — COMMENTED** [review]
Thanks for your contribution!  We found the type hints for the `symbolic` method conflicts with pytorch implementation, just like this (https://pytorch.org/docs/stable/onnx.html#):
image
Since the torch._C.xxx is a C level instance, suggest just removing the type hints for `symbolic`

**Comment by ShawnHXH** [comment]
> Thanks for your contribution! We found the type hints for the `symbolic` method conflicts with pytorch implementation, just like this (https://pytorch.org/docs/stable/onnx.html#): image Since the torch._C.xxx is a C level instance, suggest just removing the type hints for `symbolic`

Hi, the change has been made, please have a review.

File: mmcv/ops/correlation.py [hunk 1]
```diff
@@ -15,14 +17,14 @@ class CorrelationFunction(Function):
...
-                max_displacement=1,
-                stride=1,
-                padding=1,
-                dilation=1,
-                dilation_patch=1):
+                input1: Tensor,
+                input2: Tensor,
+                kernel_size: int = 1,
+                max_displacement: int = 1,
+                stride: int = 1,
```
Comment (HAOCHENYE) [inline mmcv/ops/correlation.py:24]: Should this stride be consistent with

Comment (ShawnHXH) [reply mmcv/ops/correlation.py:24]: This `stride` should be consistent with the one in `class Correlation` at the same file, which is defined as `int`.

Selection_001

**Review by HAOCHENYE — APPROVED** [review]

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**